src/Windowing/OpenToolkit.Windowing.Common/Interfaces/INativeWindow.cs

```diff
@@ -25,6 +27,215 @@ public interface INativeWindow : INativeWindowProperties, IWindowEvents, IWindow
+
+        /// <summary>
+        /// Gets or sets the current monitor.
+        /// </summary>
+        Monitor CurrentMonitor { get; set; }
+
+        /// <summary>
+        /// Gets a value representing the current graphics API.
+        /// </summary>
+        ContextAPI API { get; }
+
+        /// <summary>
+        /// Gets a value representing the current graphics API profile.
+        /// </summary>
+        ContextProfile Profile { get; }
+
+        /// <summary>
+        /// Gets a value representing the current graphics profile flags.
+        /// </summary>
+        ContextFlags Flags { get; }
+
+        /// <summary>
+        /// Gets a value representing the current version of the graphics API.
+        /// </summary>
+        Version APIVersion { get; }
+
+        /// <summary>
+        /// Gets or sets the title of the window.
+        /// </summary>
+        string Title { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether this window has input focus.
+        /// </summary>
+        /// <remarks>
+        /// A window cannot be manually unfocused by setting this to false.
+        /// </remarks>
+        bool IsFocused { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether the window is visible.
+        /// </summary>
+        bool IsVisible { get; set; }
+
+        /// <summary>
+        /// Gets or sets the <see cref="WindowState" /> for this window.
+        /// </summary>
+        WindowState WindowState { get; set; }
+
+        /// <summary>
+        /// Gets or sets the <see cref="WindowBorder" /> for this window.
+        /// </summary>
+        WindowBorder WindowBorder { get; set; }
+
+        /// <summary>
+        /// Gets or sets a <see cref="OpenToolkit.Mathematics.Box2i" /> structure the contains the external bounds of this window,
+        /// in screen coordinates.
+        /// External bounds include the title bar, borders and drawing area of the window.
+        /// </summary>
+        Box2i Bounds { get; set; }
+
+        /// <summary>
+        /// Gets or sets a <see cref="OpenToolkit.Mathematics.Vector2i" /> structure that contains the location of this window on the
+        /// desktop.
+        /// </summary>
+        Vector2i Location { get; set; }
+
+        /// <summary>
+        /// Gets or sets a <see cref="OpenToolkit.Mathematics.Vector2i" /> structure that contains the external size of this window.
+        /// </summary>
+        Vector2i Size { get; set; }
+
+        /// <summary>
+        /// Gets or sets the horizontal location of this window on the desktop.
+        /// </summary>
+        int X { get; set; }
+
+        /// <summary>
+        /// Gets or sets the vertical location of this window on the desktop.
+        /// </summary>
+        int Y { get; set; }
+
+        /// <summary>
+        /// Gets or sets the external width of this window.
+        /// </summary>
+        int Width { get; set; }
```

Comment:
How does this differ from the `Vector2i Size` ?

Do we need both of these? If not, can we drop the Width/Height and keep the vector?

Reply:
Personally I'm for removing this too. I kept it because it was already there only.